### DISCUSSION TITLE
Transparent Scientific Visualization

### DISCUSSION THREAD

**Discussion by FelixGirard** [iTwin/itwinjs-core#6212] [Q&A]
I've been able to use the scientific visualization to display what I want (basically a 3d heatmap). 
But I would like to be able to modify its transparency. 
I've tried using custom transparent colors for the [ThematicGradientSettings](https://www.itwinjs.org/reference/core-common/symbology/thematicgradientsettings/), but that didn't work (still appears solid).

It's not graphics on an iModel element, just a [Polyface](https://www.itwinjs.org/reference/core-geometry/polyface/polyface/) living in a [decorator](https://www.itwinjs.org/reference/core-frontend/views/decorator/).

Is there a way to achieve this? 
If not, could a new transparency parameter be added to the [ThematicGradientSettings](https://www.itwinjs.org/reference/core-common/symbology/thematicgradientsettings/)?
 
 Thanks!

**Comment by markschlosseratbentley** [comment]
`colorMix` on the ThematicGradientSettings is a possible solution, but that would only mix the thematic display colorization with terrain and point clouds.  It mixes the thematic colorization of the scene based on a percentage with the terrain / point clouds in the scene.

It _may_ be possible to modify the iTwin.js source code to apply `colorMix` (or some other value) to the entire scene (or maybe specific things based on some other flag), but that currently is not possible.  This is work that would have to be scheduled and merged into the iTwin.js main repo.

**Reply by FelixGirard** [reply]
For more details, I am using the analysisStyle, not thematic display for this time. `view.displayStyle.settings.analysisStyle`
I was inspired by the [scientific visualization sample](https://www.itwinjs.org/sandboxes/iTwinPlatform/Scientific%20Visualization).

`colorMix` wont work in my case because what I would like to see is under the mesh.
I think it would be a good feature to schedule. I know others in Bentley want this as well.

**Comment by pmconne** [comment]
@markschlosseratbentley does ordinary thematic display (not scientific visualization) respect the transparency values of the gradient keys? e.g., if I define a thematic gradient with one opaque color and one transparent color, will that be reflected in the geometry I apply the gradient to?

For scientific visualization, the thematic gradient comes `Target.analysisTexture`. `SurfaceGeometry.getPass` does not consider the transparency of that texture when computing whether the surface should draw in the opaque or translucent pass (or both, if gradient contains a mix of transparent and opaque colors) - so AFAICT it will render in whatever pass(es) the base geometry would render in.

**Comment by pmconne** [comment]
> so AFAICT it will render in whatever pass(es) the base geometry would render in.

Confirmed by editing the sandbox to set the decoration to 50% transparency and the render mode to smooth. Could be relatively easy to fix.
![image](https://github.com/iTwin/itwinjs-core/assets/22944042/1da16e1d-39ef-4aa5-b55c-4e0ddedbc1e9)

**Comment by markschlosseratbentley** [comment]
> @markschlosseratbentley does ordinary thematic display (not scientific visualization) respect the transparency values of the gradient keys? e.g., if I define a thematic gradient with one opaque color and one transparent color, will that be reflected in the geometry I apply the gradient to?

Ordinary thematic display ignores alpha values on the RGBA entries.  Instead, it just uses the underlying `baseColor.a` value.

**Comment by markschlosseratbentley** [comment]
Created issue: https://github.com/iTwin/itwinjs-core/issues/6241

**Comment by pmconne** [comment]
PR #6250 has merged, adding support for transparency in scientific visualization and thematic display.

**Reply by FelixGirard** [reply]
Awesome work! (And really fast too!) Thanks :)

I think we should add that to the sample later.
